lib/posix/options/CMakeLists.txt

```diff
@@ -9,10 +9,6 @@ zephyr_syscall_header_ifdef(CONFIG_POSIX_CLOCK_SELECTION posix_clock.h)
 zephyr_syscall_header_ifdef(CONFIG_POSIX_TIMERS posix_clock.h)
 zephyr_syscall_header_ifdef(CONFIG_XSI_SINGLE_PROCESS posix_clock.h)
 
-if(CONFIG_POSIX_SYSTEM_INTERFACES)
```

Comment:
The majority of this commit would seem at first to not depend on the rest of the PR, but adds lots of file changes, complicating the review.
It was already possible to use this headers without using `zephyr/posix/` in the `#include` this change before this commit by enabling this option. But this commit is removing it before the other relevant changes in this RP. And the next commit is telling 
```
This change removes the requirement to have `<zephyr/posix/...>` prefixing
    all standard POSIX includes in Zephyr.
```
Is this bisectable?
It would have been much nicer if this would have been done in a separate PR to ease review and ensure it worked.

Reply:
> It would have been much nicer if this would have been done in a separate PR to ease review and ensure it worked.

@aescolar - it was (see #97152). It was there for nearly 3 weeks, was passing CI, and you were a reviewer on it. I also asked for reviewers multiple times in Discord.

Also mentioned in the PR description, [here](https://github.com/zephyrproject-rtos/zephyr/pull/97855#issue-3529013860).

Reply:
Thanks ... I guess? I mean, it was bisectable, as the build was functional with every commit, which is at least a partial definition of bisectabiity.

It's also not good practice to ignore reviews for weeks and then bombard authors with change requests on the day of the feature freeze 🤷‍♂️